tensorflow/lite/micro/kernels/xtensa_hifimini/svdf.cc

```diff
@@ -0,0 +1,380 @@
+    ae_p24x2s dstate, dtmp, dout;
+
+    AE_LP16X2F_IU(dtmp, old_state_start, 4);
+    AE_LP16X2F_IU(dstate, old_state_start, 4);
+    for(int i=0; i<(loopcnt>>1); i++) {
+      dout = AE_SELP24_LH(dtmp, dstate);
+      dtmp = dstate;
+      AE_LP16X2F_IU(dstate, old_state_start, 4);
+      AE_SP16X2F_IU(dout, new_state_start, 4);
+    }
+    if(loopcnt & 0x1)
+    {
+      AE_SP16F_L_I(dtmp, (ae_p16s*)new_state_start, 4);
+    }
+  }
+  else
+  {
+    // 2-bytes aligned processing
+    ae_p16s* new_state_start = (ae_p16s*)(state_ptr-1);
+    const ae_p16s* old_state_start = (ae_p16s*)(state_ptr);
+    int loopcnt = (n_batch * n_filter * n_memory)-1;
+    ae_p24x2s dstate;
+    for(int i=0; i<loopcnt; i++) {
+      AE_LP16F_IU(dstate, old_state_start, 2);
+      AE_SP16F_L_IU(dstate, new_state_start, 2);
+    }
+  }
+  // Note: no need to clear the latest activation, matmul is not accumulative.
+
+  // Feature matmul.
+  {
+    int16_t* state = GetTensorData<int16_t>(activation_state_tensor);
+    const int8_t* input = GetTensorData<int8_t>(input_tensor);
+    const int8_t* weight_feature =
+        GetTensorData<int8_t>(weights_feature_tensor);
+    int16_t* result_in_batch = state + (n_memory - 1);
+    int err = 0;
+
+    for (int b = 0; b < n_batch; b++) {
+      err = xa_nn_matXvec_out_stride_sym8sxasym8s_16(&result_in_batch[b*n_filter*n_memory],
+                                                     weight_feature,
+                                                     &input[b*n_input],
+                                                     NULL,
+                                                     n_filter,
+                                                     n_input,
+                                                     n_input,
+                                                     n_memory,
+                                                     -input_zp,
+                                                     (data.effective_scale_1_a<<8),
+                                                     data.effective_scale_1_b);
+      CHECK_ERR_HIFI_NNLIB_KER(err, "xa_nn_vec_matXvec_sym8sxasym8s_16 failed");
+    }
+  }
+
+  // Time.
+  {
+    for (int b = 0; b < n_batch; ++b) {
+      int8_t* output_ptr =
+          GetTensorData<int8_t>(output_tensor) + b * n_unit;
+
+      const int16_t* vector1_ptr = GetTensorData<int16_t>(weights_time_tensor);
+      const int16_t* vector2_ptr =
+          GetTensorData<int16_t>(activation_state_tensor) +
+          b * n_memory * n_filter;
+      int err = 0;
+      const int32_t *bias_ptr = GetTensorData<int32_t>(bias_tensor);
+      err = xa_nn_dot_prod_16x16_asym8s(output_ptr,
+          vector1_ptr,
+          vector2_ptr,
+          bias_ptr,
+          n_memory*n_rank,
+          (data.effective_scale_2_a<<8),
+          data.effective_scale_2_b,
+          output_zp,
+          n_unit);
+      CHECK_ERR_HIFI_NNLIB_KER(err, "xa_nn_dot_prod_16x16_asym8s failed");
```

Comment:
`New CHECK or LOG(FATAL) statements inserted in code! Prefer using DCHECK, LOG(DFATAL), or DLOG(FATAL) instead`

Reply:
CHECK_ERR_HIFI_NNLIB_KER is replaced with TF_LITE_KERNEL_LOG